python/desc/sims/GCRCatSimInterface/CatalogClasses.py

```diff
@@ -4,16 +4,66 @@
 from lsst.utils import getPackageDir
 from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogSersic2D
 from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogZPoint
+from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogSN
 from lsst.sims.catUtils.mixins import VariabilityAGN
 from lsst.sims.catalogs.decorators import cached, compound
 from lsst.sims.catUtils.mixins import EBVmixin
 
-__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN"]
+
+__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN", "DC2PhosimCatalogSN"]
 
 #########################################################################
 # define a class to write the PhoSim catalog; defining necessary defaults
 
 
+class DC2PhosimCatalogSN(PhoSimCatalogSN):
+    """
+    Modification of the PhoSimCatalogSN mixin to provide shorter sedFileNames
+    by leaving out the parts of the directory name. Also fix name changes from
+    gamma to shear.
+    """
+    def get_sedFilepath(self):
+        return self.column_by_name('TsedFilepath')
+
+    def get_shorterFileNames(self):
+        """
+        Method to truncate filenames for transient
+        spectra written out by phosim. 
+
+        .. note: the variable sep needs to be in
+        `self.sn_sedfile_prefix` before writing out
+        a phosim catalog.
+        """
+        fnames = self.column_by_name('sedFilepath')
+        sep = 'specFileSN_'
+        split_names = []
+        for fname in fnames:
+            if 'None' not in fname:
+                fname = sep + fname.split(sep)[-1] 
+            else:
+                fname = 'None'
+            split_names.append(fname)
+        return np.array(split_names)
+
+    # column_outputs = PhoSimCatalogSN.column_outputs
+    # column_outputs[PhoSimCatalogSN.column_outputs.index('sedFilepath')] = \
+    #    'shorterFileNames'
+    column_outputs = ['prefix', 'uniqueId', 'raPhoSim', 'decPhoSim',
+                      'phoSimMagNorm', 'shorterFileNames', 'redshift',
+                      'shear1', 'shear2', 'kappa', 'raOffset', 'decOffset',
```

Comment:
I just want to explicitly state that we are not going to be applying weak lensing parameters to these supernovae, since there are no weak lensing parameters in `SNFileDBObject`.  `shear1`, `shear2`, and `kappa` will always return their default values of 0.0 in the case of supernovae.

Reply:
- As I said I am happy to remove them. If there is a suggestion of where to track such products, I will follow that. Or else keep these in a separate repository for my own tracking.

- About weak lensing: Yes, I know about this. I have been thinking that I would go with unlensed SN in Run 1.1 and make sure that the SN land where I put them wrt the protoDC2 galaxies. My guess is that if I provide these values in the csv files, then it would be trivial to turn them on in Run 1.2, at least with the shear values provided in protoDC2.